overlord/snapstate/snapmgr.go

```diff
@@ -224,3 +228,20 @@ func (m *SnapManager) Wait() {
 func (m *SnapManager) Stop() {
 	m.runner.Stop()
 }
+
+// SnapInfo returns snap.Info for a given snap name and snap version.
+func SnapInfo(state *state.State, snapName, snapVersion string) (*snap.Info, error) {
```

Comment:
It's important to understand why @pedronis requested it. The logic below takes information out of the snap metadata right now, but the information that comes within the snap blob is limited. Things like Developer, Revision, etc, won't be inside it. We'll store that information in the state.